src/mlpack/methods/reinforcement_learning/sac.hpp

```diff
@@ -93,6 +93,16 @@ class SAC
       UpdaterType policyNetworkUpdater = UpdaterType(),
       EnvironmentType environment = EnvironmentType());
 
+  /**
+   * Delete copy constructor.
+   */
+  SAC(SAC const&) = delete;
```

Comment:
Are you sure we want to delete the copy constructor?  Would anyone ever want to copy a `SAC` object?

Reply:
I am not sure, but I am not able to copy the `SAC` object since there is no delclared copy constructor.

Reply:
Ahh, maybe we should add one of those too?  It unfortunately can't be the default because of the `qNetworkUpdatePolicy` and `policyNetworkUpdatePolicy` members.  (Also, I think we can remove those `#ifdef`s because the minimum version of ensmallen is 2.10.0.)

Reply:
Agreed, it will be much better to add a copy constructor.
I will add one, and remove all these `#ifdef `